tests/lib/c_lib/common/src/main.c

```diff
@@ -15,10 +15,6 @@
  * it guarantee that ALL functionality provided is working correctly.
  */
 
-#ifdef CONFIG_NEWLIB_LIBC
```

Comment:
Are you sure it is a good idea to remove this?
We will also need to set it for the host libC:
https://github.com/zephyrproject-rtos/zephyr/pull/67052/files#diff-245b0fca9de29e79ca821147f72ef70ce13d8a855bfa80e9c53b304697090f30

Reply:
Yes, the current Zephyr C library standard includes all of the functions used in this file, so there should be no additional defines required. If the C library in use does not expose the full Zephyr C library API, then that's a bug in the library integration which should be solved there and not in the test.

Reply:
Another kind of degree of freedom we wanted though was to allow sources to `#define _POSIX_C_SOURCE <ver>` at the top of the file.

It should be reasonable to leave in if that's preferred, but it might be better to do something like this

```cpp
#if !defined(_POSIX_C_SOURCE) || _POSIX_C_SOURCE < [NEEDED_VER]
#undef _POSIX_C_SOURCE
#define _POSIX_C_SOURCE [NEEDED_VER]
#endif
```